.github/workflows/release.yml

```diff
@@ -94,6 +114,7 @@ jobs:
   cache-stack:
     name: 'Cache Stack'
     runs-on: ubuntu-22.04
+    needs: draft-release
```

Comment:
Ditto, this job creates a cache that should be used by all PRs so it must run on the `master` branch.